File: previous_versions/python/py-files/ascii_read.py
##------------------------------------------------------
## Read input DEM stored in ../input 
##------------------------------------------------------
import numpy
# -----------------------
# read input ascii file
#------------------------
input_dem = '../input/dem.asc'

fid = open(input_dem,'r')

if fid == -1:
     raise Exception("Cannot find file")
 
# Read header information and store in structure
for i in range(1,7):
    tline    = fid.readline()
    data_str = tline.partition(' ')[2]
#     print data_str
    if i == 1: 
       ncols         = int(data_str)
    elif i == 2:
        nrows         = int(data_str)
    elif i == 3:
        xllcorner     = float(data_str)
    elif i == 4:  
        yllcorner     = float(data_str)
    elif i == 5:
        cellsize      = float(data_str)
    elif i == 6:
        NODATA_value  = int(data_str)
 
# Read dataset

A = numpy.zeros([nrows, ncols], dtype=float)

for i in xrange(nrows):
            row = numpy.zeros([ncols])    
            line = fid.readline()
 
            for j, val in enumerate(line.split(), start = 0):
                value = float(val)
                if value == NODATA_value:
                    #value = Nan
                    value = numpy.nan
                row[j] = value
            A[i,:] = row
fid.close()

A = A.conj().transpose()
max_data = A.max()
nan_ind = numpy.isnan(A)

A[nan_ind] = max_data
NODATA_value = max_data

data = A

File: previous_versions/octave/m-files/ascii_read.m
function data_struct = ascii_read(fname)


% Open file for reading
  fid = fopen(fname,'r');

 if fid == -1 
     error(['Cannot find file']);
     return
 end

% Read header information and store in structure
for i = 1:6
    tline    = fgetl(fid);
    data_str = str2num(tline(index(tline,' '):end)); 
    if i == 1
        data_struct.ncols         = data_str;
    elseif i == 2
        data_struct.nrows         = data_str;
    elseif i == 3
        data_struct.xllcorner     = data_str;
    elseif i == 4    
        data_struct.yllcorner     = data_str;
    elseif i == 5
        data_struct.cellsize      = data_str;
    elseif i == 6
        data_struct.NODATA_value  = data_str;
    end
end

% Read data
A = fscanf(fid,'%f',[data_struct.ncols,data_struct.nrows]);

% Close file
fclose(fid);

A = A';

 A(A == data_struct.NODATA_value) = NaN;

 max_data = max(max(A));
 nan_ind = isnan(A);

 A(nan_ind) = (max_data);
 data_struct.NODATA_value = (max_data);

% data_struct.data        = flipud(A);
 data_struct.data        = A;

endfunction





